File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        
        HashMap<String, Integer> list = new HashMap<>();
        
        for(int i=0; i<name.length; i++){
            list.put(name[i], yearning[i]);
        }
        
        int[] answer = new int[photo.length];
        
        for(int i=0; i<photo.length; i++){
            int sum = 0;
            for(int j=0; j<photo[i].length; j++){
                sum += list.getOrDefault(photo[i][j], 0);
            }
            answer[i] = sum;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/176963. 추억 점수/추억 점수.js
function solution(name, yearning, photo) {
    
    const list = new Map();
    
    for(let i=0; i<name.length; i++){
        list.set(name[i], yearning[i]);
    }
    
    var answer = [];
    
    for(let i=0; i<photo.length; i++){
        let sum = 0;
        for(let j=0; j<photo[i].length; j++){
            if(list.has(photo[i][j])){
                sum += list.get(photo[i][j]);
            }
        }
        answer.push(sum);
    }
    
    return answer;
}